docs/merkletree/treetex/main.go

```diff
@@ -176,14 +176,19 @@ func perfectMega(prefix string, height uint, leafIndex uint64) {
 }
 
 // perfect renders a perfect subtree.
-func perfect(prefix string, height uint, index uint64, nodeText nodeTextFunc) {
-	perfectInner(prefix, height, index, true, nodeText)
+func perfect(prefix string, height uint, index uint64, nodeText, dataText nodeTextFunc) {
+	perfectInner(prefix, height, index, true, nodeText, dataText)
 }
 
 // drawLeaf emits TeX code to render a leaf.
-func drawLeaf(prefix string, index uint64, text string) {
+func drawLeaf(prefix string, index uint64, leafText, dataText string) {
 	a := nInfo[compact.NewNodeID(0, index)]
-	fmt.Printf("%s [%s, %s, align=center, tier=leaf]\n", prefix, text, a.String())
+
+	// First render the leaf node of the Merkle tree
+	fmt.Printf("%s [%s, %s, align=center, tier=leaf\n", prefix, leafText, a.String())
+	// and then a child-node representing the leaf data itself:
+	a.leaf = true
```

Comment:
How come it was false before this change?

Reply:
I'm pretty sure it should be `true` already. Can you assert it locally?

Reply:
Ah, that's because I removed the bit where it was set to true inside `perfectInner` - arguably it makes more sense to just handle this here since it's really about presentation.

wdyt?

Reply:
Yeah, I think it's better to handle presentation stuff here, as opposed to scattering it between functions.